File: ShopApp_Backend/Shopapp/src/main/java/com/project/Shopapp/Models/SanPham.java
package com.project.Shopapp.Models;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "SANPHAM")
@Builder
public class SanPham extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int MASANPHAM;

    @Column(name = "TENSANPHAM", nullable = false)
    private String TENSANPHAM;

    @Column(name = "GIA", nullable = false)
    private int GIA;

    @ManyToOne
    @JoinColumn(name = "MATHUONGHIEU")
    private ThuongHieu MATHUONGHIEU;

    private String MOTA;
    private int SOLUONGTONKHO;

    @ManyToOne
    @JoinColumn(name = "MALOAISANPHAM")
    private LoaiSanPham MALOAISANPHAM;

    private String THUMBNAIL;
}

File: webadmin/src/app/models/sanpham.ts
import { HinhAnh } from "./hinhanh.image";

export interface SanPham {
    masanpham:number;
    tensanpham: string;
    gia: number;
    mathuonghieu: number;
    mota: string;
    soluongtonkho: number;
    maloaisanpham: number;
    thumbnail_url:string;
    thumbnail:string;
    hinhAnhUrls: HinhAnh[];
}